src/java/org/apache/cassandra/index/sai/utils/IndexTermType.java

```diff
@@ -141,20 +141,24 @@ private IndexTermType(ColumnMetadata columnMetadata, List<ColumnMetadata> partit
         this.indexTargetType = indexTargetType;
         this.capabilities = calculateCapabilities(columnMetadata, partitionColumns, indexTargetType);
         this.indexType = calculateIndexType(columnMetadata.type, capabilities, indexTargetType);
-        if (indexType.subTypes().isEmpty())
+
+        AbstractType<?> baseType = indexType.unwrap();
```

Comment:
This bit isn't part of the issue described in the Jira, but it's just...clearly wrong, so I cleaned it up.